modules/tinymce/src/core/test/ts/browser/dom/BundledCssTest.ts

```diff
@@ -0,0 +1,256 @@
+  const skinUiContentJs = (skin: 'oxide' | 'oxide-dark', inline: boolean) => skinUiDir(skin) + `content${inline ? '.inline' : ''}.js`;
+
+  const styleTagsToKeys = (styleTags: SugarElement<HTMLStyleElement>[]) => {
+    return Arr.map(styleTags, (tag) => tag.dom.dataset.mceKey);
+  };
+
+  const linkTagsToKeys = (linkTags: SugarElement<HTMLLinkElement>[], basePath: string) => {
+    return Arr.map(linkTags, (tag) => tag.dom.href.replace(basePath, ''));
+  };
+
+  const getCssTags = (container: SugarElement<HTMLHeadElement | ShadowRoot>) => {
+    const style =
+      SelectorFilter.descendants<HTMLStyleElement>(
+        container,
+        'style'
+      );
+    const link =
+      SelectorFilter.descendants<HTMLLinkElement>(
+        container,
+        'link'
+      );
+
+    return {
+      style,
+      link
+    };
+  };
+
+  const getIframeCSSTags = (editor: Editor) =>
+    getCssTags(SugarShadowDom.getStyleContainer(SugarShadowDom.getRootNode(TinyDom.body(editor))));
+
+  const getCSSContainerTags = (editor: Editor) =>
+    getCssTags(SugarShadowDom.getStyleContainer(SugarShadowDom.getRootNode(TinyDom.targetElement(editor))));
+
+  const pCreateEditor = async (shadowDom: boolean, settings: Record<string, any>) => {
+    if (shadowDom) {
+      const shadowHost = SugarElement.fromTag('div', document);
+      Insert.append(SugarBody.body(), shadowHost);
+      const sr = SugarElement.fromDom(shadowHost.dom.attachShadow({ mode: 'open' }));
+      const editorDiv = SugarElement.fromTag('div', document);
+      Insert.append(sr, editorDiv);
+
+      const cleanup = () => {
+        Remove.remove(shadowHost);
+      };
+      const editor = await McEditor.pFromElement<Editor>(editorDiv, {
+        ...settings
+      });
+      return [ editor, cleanup ] as const;
+    } else {
+      const editor = await McEditor.pFromSettings<Editor>({
+        ...settings
+      });
+      const cleanup = Fun.noop;
+      return [ editor, cleanup ] as const;
+    }
+  };
+
+  for (const inline of [ false, true ]) {
+    context('inline: ' + inline, () => {
+      for (const shadowdom of [ false, true ]) {
+        context('shadowdom: ' + shadowdom, () => {
+          for (const contentCss of [ 'default', 'dark' ] as const) {
+            context('contentCSS: ' + contentCss, () => {
+              for (const skin of [ 'oxide', 'oxide-dark' ] as const) {
+                context('skin: ' + skin, () => {
+                  const oxideUiSkinMap: Record<string, string> = {
+                    'oxide-dark': 'dark',
+                    'oxide': 'default',
+                    'tinymce-5': 'tinymce-5',
+                    'tinymce-5-dark': 'tinymce-5-dark'
+                  };
+
+                  context('no bundling', () => {
+                    it('TINY-11558: Load CSS styesheets as links', async () => {
+                      const [ editor, cleanup ] = await pCreateEditor(
```

Comment:
What is the convention behind naming specific variables like this with `p`? 
Is it short for something like procedure? I'd be curious to know when or where we should use it!

Reply:
The `p` prefix indicates that the function returns a promise. We only use that convention in tests and it mainly makes it easier to know when we need to use `await` etc.